content/documents/working-with-specs.mdx

```diff
@@ -0,0 +1,210 @@
+
+## Spec Types
+
+The configuration of a spec defines how a user will interact with a product it has been assigned to. Specs can generally be categorised under one of the following types:
+
+### Open Text Spec
+
+The `AllowOpenText` property provides the user the option of inputting text associated to the product, e.g. a name to be engraved onto a pen.
+
+<CodeExample
+  title="Creating an open text spec"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/specs HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8;\n
+{
+    "ID": "OPEN_TEXT_SPEC",
+    "Name": "Open Text Spec",
+    "AllowOpenText": true
+}`
+}}
+/>
+
+### Options Spec
+
+To give the user the ability to modify a product configuration while still limiting their selection, a spec with options can be created, e.g. business cards may have a number of design templates to choose from.
+
+<CodeExample
+  title="Creating an options spec"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/specs HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8;\n
+{
+    "ID": "OPTIONS_SPEC",
+    "Name": "Options Spec"
+}`
+}}
+/>
+
+#### Spec Options
+
+Spec options represent the available options that can be selected for an option spec.
+
+<CodeExample
+  title="Creating a spec option"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/specs/PREDEFINED_OPTIONS_SPEC/options HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8;\n
+{
+    "ID": "SPEC_OPTION_1",
+    "Name": "Spec Option 1"
+}`
+}}
+/>
+
+### Variant Spec
+
+A variant spec is the option spec with the difference that it slices a product along each spec option to create a unique variant product per option, e.g. a television with different sizes each representing a variant product.
+
+<CodeExample
+  title="Creating a variant spec"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/specs HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8;\n
+{
+    "ID": "VARIANT_SPEC",
+    "Name": "Variant Spec",
+    "DefinesVariant": true,
+    "Required": true
+}`
+}}
+/>
+
+>Variant specs must be marked as `Required` as the product object by itself no longer represents a unique product, but as housing for related variant products and cannot be added to an order without specifying a specific variant product.
+
+#### Generate Variants
+
+Once spec product assignments have been defined for a product, the variants can then be generated under the products resource.
+
+<CodeExample
+  title="Generating product variants"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/products/INSERT_PRODUCT_ID_HERE/variants/generate  HTTP/1.1
```

Comment:
Can we use the same format as the API reference does? So instead of `INSERT_PRODUCT_ID_HERE` we just use `{productID}`

Reply:
Again, I believe I was keeping consistency with Learn docs, but can update and add to documentation guidelines/conventions.